sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -300,8 +318,12 @@ private[joins] class UnsafeHashedRelation(
   }
 
   override def read(kryo: Kryo, in: Input): Unit = Utils.tryOrIOException {
-    read(() => in.readInt(), () => in.readLong(), in.readBytes)
+    read(() => in.readBoolean(), () => in.readInt(), () => in.readLong(), in.readBytes)
   }
+
+  override def inputEmpty: Boolean = binaryMap.inputEmpty
```

Comment:
Don't need `!anyNullKeyExists` here?

Reply:
anyNullKeyExists already inlined in binaryMap.inputEmpty